ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -120,23 +120,35 @@ def main(argv=sys.argv[1:]):
     parser.add_argument(
         '--outputdir', type=str, default='api',
         help='Output directory')
+    parser.add_argument(
+        '--packages-select',
+        default=[],
+        nargs='*',
+        help='Generate the documentation for the following package names')
     args = parser.parse_args(argv)
 
     html_dir = os.path.join(args.outputdir, 'html')
     os.makedirs(html_dir, exist_ok=True)
 
-    messages = get_message_interfaces()
-    services = get_service_interfaces()
-    actions = get_action_interfaces()
+    try:
+        messages = get_message_interfaces(args.packages_select)
```

Comment:
Does this API do the right thing when passing an empty list?

Reply:
when you call this function with an empty list then all the packages are returned